### ISSUE TITLE
i can't use TabNine

### ISSUE THREAD

**Issue by zz541843** [codota/TabNine#427]
![2](https://user-images.githubusercontent.com/72638366/127274835-24020309-2aed-40c6-8b31-0aece88f7c94.png)
The ide of this computer is the latest version, but repeated installation for several times has no effect, and the setting of tabnine cannot be opened. Just like it was not installed, it was successfully installed on another computer in the same way. I don't know why

gz#4357

**Comment by zz541843** [comment]
I don't know why, this process keeps increasing itself until the memory overflows

**Comment by Eransho** [comment]
Hi @zz541843 , can you share your installation info (you can find them on Tabnine Hub) and your tabnine log?

**Comment by maya-tabnine** [comment]
To get Tabnine logs, you need to locate your binary files and run the engine directly.
On Tabnine Hub, scroll down to the Engine file path, and find the Tabnine file path.

Try to run the following command from the terminal using the right path:
`PATH_TO_ENGINE/TabNine.cmd --log_to_stderr`
Would you please collect the output to a file and share it with us?

And to attach your config file with the installation info, you can locate it on Tabnine Hub, scroll down to the Config file path, and attach the file.

Thanks!

**Comment by zz541843** [comment]
@maya-tabnine  Thank for you reply ! Where is TabNine? And I didn't find tabnine.cmd on the computer. I have forcibly uninstalled tabnine. After restarting the computer, it is still invalid.Here are some information I know
As soon as I click tabnine in webstorm, he will frantically start a new process, and the generated files are in this directory
![3](https://user-images.githubusercontent.com/72638366/127417372-3657276e-9fa9-42e9-bace-63efe3e32e18.png)
This directory may be the installation directory of tabnine
![4](https://user-images.githubusercontent.com/72638366/127417373-cf6a00db-3bfb-49bd-b931-977e027265b9.png)

**Comment by maya-tabnine** [comment]
You can reach Tabnine Hub by typing "tabnine::config" in the IDE editor or click the tabnine on the status bar if it exists.
Let me know it works for you!

**Comment by zz541843** [comment]
> You can reach Tabnine Hub by typing "tabnine::config" in the IDE editor or click the tabnine on the status bar if it exists.
> Let me know it works for you!

Tabnine:: config and clicking the IDE icon are useless, which is one of the main reasons why I come to help

**Comment by zz541843** [comment]
> You can reach Tabnine Hub by typing "tabnine::config" in the IDE editor or click the tabnine on the status bar if it exists.
> Let me know it works for you!

And both IDEA and GoLand have tabnine installed, which is also invalid. They are the latest version

**Comment by yonip23** [comment]
Hi @zz541843 
Can you please run `C:\Users\<user>\.tabnine\3.5.31\x86_64-pc-windows-gnu\TabNine.exe --log_to_stderr`for a few seconds and attach here the output from your console?

**Comment by zz541843** [comment]
> Hi @zz541843
> Can you please run `C:\Users\<user>\.tabnine\3.5.31\x86_64-pc-windows-gnu\TabNine.exe --log_to_stderr`for a few seconds and attach here the output from your console?
[2021-08-02T16:43:44.648150400+08:00][INFO][24220 main][auth::token][Some("shared/auth/src/token/mod.rs")][Some(41)] token path: C:\Users\26735\AppData\Roaming\TabNine\tabnine.token
[2021-08-02T16:43:44.656503800+08:00][WARN][24220 main][auth::user][Some("shared/auth/src/user/mod.rs")][Some(442)] No user token available: 系统找不到指定的文件。 (os error 2)
[2021-08-02T16:43:44.658856900+08:00][INFO][24220 main][TabNine::main_utils][Some("tabnine/frontend/src/main_utils.rs")][Some(182)] TabNine.exe --log_to_stderr
[2021-08-02T16:43:44.762634700+08:00][INFO][24220 main][watchdog_api::wd_client][Some("shared/wd_api/src/wd_client.rs")][Some(74)] Initializing watchdog client...
[2021-08-02T16:43:45.032527700+08:00][INFO][24220 main][watchdog_api::wd_client][Some("shared/wd_api/src/wd_client.rs")][Some(216)] Starting watchdog...
[2021-08-02T16:43:45.036417300+08:00][INFO][24220 bs_main_autobarker][watchdog_api::wd_client][Some("shared/wd_api/src/wd_client.rs")][Some(216)] Starting watchdog...
[2021-08-02T16:43:45.044757600+08:00][INFO][24220 bootstrapper_main_run_loop][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(196)] trying to start tabnine version Some("3.5.31")
[2021-08-02T16:43:45.046154+08:00][INFO][24220 bootstrapper_main_run_loop][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(287)] spawning tabnine with args ["TabNine.exe", "--log_to_stderr", "--no_bootstrap"]
[2021-08-02T16:43:45.048926500+08:00][INFO][24220 bootstrapper_main_run_loop][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(291)] started tabnine process pid 12236 version Some("3.5.31"), desired version was Some("3.5.31") this version is Some("3.5.31")
[2021-08-02T16:43:45.059731800+08:00][INFO][12236 main][auth::token][Some("shared/auth/src/token/mod.rs")][Some(41)] token path: C:\Users\26735\AppData\Roaming\TabNine\tabnine.token
[2021-08-02T16:43:45.061729600+08:00][WARN][12236 main][auth::user][Some("shared/auth/src/user/mod.rs")][Some(442)] No user token available: 系统找不到指定的文件。 (os error 2)
[2021-08-02T16:43:45.062738900+08:00][INFO][12236 main][TabNine::main_utils][Some("tabnine/frontend/src/main_utils.rs")][Some(182)] TabNine.exe --log_to_stderr --no_bootstrap
[2021-08-02T16:43:45.063808200+08:00][INFO][12236 main][watchdog_api::wd_client][Some("shared/wd_api/src/wd_client.rs")][Some(74)] Initializing watchdog client...
[2021-08-02T16:43:45.292447900+08:00][INFO][12236 main][watchdog_api::wd_client][Some("shared/wd_api/src/wd_client.rs")][Some(216)] Starting watchdog...
[2021-08-02T16:43:45.304489900+08:00][INFO][21784 main][auth::token][Some("shared/auth/src/token/mod.rs["2021-08-02T16:43:45.305642500+08:00)][][INFOSome][(1897641 )main] ][token path: auth::tokenC:\Users\26735\AppData\Roaming\TabNine\tabnine.token][
Some("shared/auth/src/token/mod.rs")][Some(41)] token path: C:\Users\26735\AppData\Roaming\TabNine\tabnine.token
[[2021-08-02T16:43:45.310451800+08:002021-08-02T16:43:45.310451200+08:00][][WARNWARN][][1897621784  mainmain][][auth::userauth::user][][SomeSome((""shared/auth/src/user/mod.rsshared/auth/src/user/mod.rs""))][][SomeSome((442442))] ] No user token available: No user token available: 系统找不到指 定的文件。系统找不到指定的文件。 (os error  (os error 22))

[[2021-08-02T16:43:45.312841700+08:002021-08-02T16:43:45.312848100+08:00][][INFOINFO][][1897621784  mainmain][][watchdog_server::wd_analyticswatchdog_server::wd_analytics][][SomeSome((""tabnine/watchdog_server/src/wd_analytics.rstabnine/watchdog_server/src/wd_analytics.rs""))][][SomeSome((4040))] ] starting heartbeat loop for watchdogstarting heartbeat loop for watchdog

[2021-08-02T16:43:45.314882600+08:00][INFO][21784 main][WD_TabNine][Some("tabnine/watchdog_server/src/main.rs")][Some(85)] terminating because another watchdog is already running.
[2021-08-02T16:43:45.318431600+08:00][INFO][24220 watchdog spawning][process_setup::command_ext][Some("shared/process_setup/src/command_ext.rs")][Some(24)] child exited with status exit code: 0
[2021-08-02T16:43:45.381470700+08:00][INFO][18976 main][watchdog_server::monitor_loop][Some("tabnine/watchdog_server/src/monitor_loop.rs")][Some(209)] watching new process 24220-TabNine.exe
[2021-08-02T16:43:45.418374300+08:00][DEBUG][12236 main][server::in_app_messaging][Some("tabnine/server/src/in_app_messaging.rs")][Some(28)] registering in app message analytics handler
[2021-08-02T16:43:45.429740200+08:00][INFO][12236 main][lunar_connection::private_models][Some("tabnine/lunar_connection/src/private_models.rs")][Some(97)] No private model found
[2021-08-02T16:43:45.431052900+08:00][INFO][12236 main][lunar_connection::model_chooser][Some("tabnine/lunar_connection/src/model_chooser.rs")][Some(31)] current private model is None
[2021-08-02T16:43:45.434577600+08:00][INFO][12236 main][lunar_connection::model_chooser][Some("tabnine/lunar_connection/src/model_chooser.rs")][Some(33)] using initial model Model {
    model_hash_override: None,
    model_size: Small,
    base_url: "https://update.tabnine.com",
}
[2021-08-02T16:43:45.440179+08:00][INFO][12236 main][lunar_connection::model_chooser_from_experiment][Some("tabnine/lunar_connection/src/model_chooser_from_experiment.rs")][Some(21)] experiment models are []
[2021-08-02T16:43:45.440559700+08:00][INFO][12236 lunar settings update thread (Local default local)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(74)] lunar settings update thread started (Local default local)
[2021-08-02T16:43:45.440629200+08:00][INFO][12236 private-model-update-loop][lunar_connection::private_models][Some("tabnine/lunar_connection/src/private_models.rs")][Some(149)] Starting private model update thread
[2021-08-02T16:43:45.441885600+08:00][INFO][12236 main][frontend::initialize][Some("tabnine/frontend/src/initialize.rs")][Some(5)] Started.
[2021-08-02T16:43:45.442269900+08:00][INFO][12236 lunar settings update thread (Remote cloud)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(74)] lunar settings update thread started (Remote[ 2021-08-02T16:43:45.451226+08:00cloud][)INFO
][[189762021-08-02T16:43:45.445240600+08:00 ][mainINFO][][watchdog_server::monitor_loop12236][ Someprivate-model-update-loop(]["lunar_connection::private_modelstabnine/watchdog_server/src/monitor_loop.rs]["Some)(]["Sometabnine/lunar_connection/src/private_models.rs("209))][] Somewatching new process (12236-TabNine.exe158
)] active private model is None
[2021-08-02T16:43:45.465011+08:00][ERROR][12236 main][lunar_connection::cloud_availability][Some("tabnine/lunar_connection/src/cloud_availability.rs")][Some(104)] Os { code: 2, kind: NotFound, message: "系统找不到指定的文件。" }
[2021-08-02T16:43:45.566124400+08:00][INFO][24272 main][auth::token][Some("shared/auth/src/token/mod.rs")][Some(41)] token path: C:\Users\26735\AppData\Roaming\TabNine\tabnine.token
[2021-08-02T16:43:45.568705500+08:00][WARN][24272 main][auth::user][Some("shared/auth/src/user/mod.rs")][Some(442)] No user token available: 系统找不到指定的文件。 (os error 2)
[2021-08-02T16:43:45.570326900+08:00][INFO][24272 main][watchdog_server::wd_analytics][Some("tabnine/watchdog_server/src/wd_analytics.rs")][Some(40)] starting heartbeat loop for watchdog
[2021-08-02T16:43:45.572258+08:00][INFO][24272 main][WD_TabNine][Some("tabnine/watchdog_server/src/main.rs")][Some(85)] terminating because another watchdog is already running.
[2021-08-02T16:43:45.576354500+08:00][INFO][12236 watchdog spawning][process_setup::command_ext][Some("shared/process_setup/src/command_ext.rs")][Some(24)] child exited with status exit code: 0
[2021-08-02T16:43:46.277934700+08:00][INFO][24220 bootstrapper_main_run_loop][bootstrapper::bootstrapper_lib][Some("tabnine/bootstrapper/src/bootstrapper_lib.rs")][Some(256)] current version is Some(Version { major: 3, minor: 5, patch: 31, pre: [], build: [] }), spawned version is Some(Version { major: 3, minor: 5, patch: 31, pre: [], build: [] }), remote version is Version { major: 3, minor: 5, patch: 33, pre: [], build: [] }
[2021-08-02T16:43:46.318585100+08:00][INFO][24220 bootstrapper_main_run_loop][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(611)] lock is at "C:\\Users\\26735\\.tabnine\\0.0.1\\x86_64-pc-windows-gnu\\bundle.lock"
[2021-08-02T16:43:46.322982600+08:00][INFO][24220 bootstrapper_main_run_loop][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(267)] downloading from "https://update.tabnine.com/bundles/3.5.33/x86_64-pc-windows-gnu/TabNine.zip" to "C:\\Users\\26735\\.tabnine\\0.0.1\\x86_64-pc-windows-gnu\\TabNine.zip"
[2021-08-02T16:43:46.326559700+08:00][INFO][24220 bootstrapper_main_run_loop][bootstrapper::download][Some("tabnine/bootstrapper/src/download.rs")][Some(29)] parent path is "C:\\Users\\26735\\.tabnine\\0.0.1\\x86_64-pc-windows-gnu"
[2021-08-02T16:43:46.330089400+08:00][INFO][24220 bootstrapper_main_run_loop][http_requests][Some("shared/http_requests/src/lib.rs")][Some(103)] Downloading from https://update.tabnine.com/bundles/3.5.33/x86_64-pc-windows-gnu/TabNine.zip
[2021-08-02T16:43:51.355425700+08:00][INFO][24220 bootstrapper_main_run_loop][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(300)] completed downloading to "C:\\Users\\26735\\.tabnine\\0.0.1\\x86_64-pc-windows-gnu\\TabNine.zip"
[2021-08-02T16:43:51.358374700+08:00][INFO][24220 bootstrapper_main_run_loop][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(414)] outpath is "C:\\Users\\26735\\.tabnine\\0.0.1\\x86_64-pc-windows-gnu\\TabNine-deep-cloud.exe"
[2021-08-02T16:43:51.445102500+08:00][INFO][24220 bootstrapper_main_run_loop][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(414)] outpath is "C:\\Users\\26735\\.tabnine\\0.0.1\\x86_64-pc-windows-gnu\\TabNine-deep-local.exe"
[2021-08-02T16:43:51.557052800+08:00][INFO][24220 bootstrapper_main_run_loop][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(414)] outpath is "C:\\Users\\26735\\.tabnine\\0.0.1\\x86_64-pc-windows-gnu\\TabNine.exe"
[2021-08-02T16:43:51.712148+08:00][INFO][24220 bootstrapper_main_run_loop][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(414)] outpath is "C:\\Users\\26735\\.tabnine\\0.0.1\\x86_64-pc-windows-gnu\\WD-TabNine.exe"
[2021-08-02T16:43:51.781529400+08:00][INFO][24220 bootstrapper_main_run_loop][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(447)] creating root path "C:\\Users\\26735\\.tabnine\\3.5.33\\x86_64-pc-windows-gnu"
[2021-08-02T16:43:51.784864500+08:00][INFO][24220 bootstrapper_main_run_loop][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(462)] moving from "C:\\Users\\26735\\.tabnine\\0.0.1\\x86_64-pc-windows-gnu" to "C:\\Users\\26735\\.tabnine\\3.5.33\\x86_64-pc-windows-gnu"
[2021-08-02T16:43:51.788828200+08:00][INFO][24220 bootstrapper_main_run_loop][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(507)] moving "C:\\Users\\26735\\.tabnine\\0.0.1\\x86_64-pc-windows-gnu\\bundle.lock" to "C:\\Users\\26735\\.tabnine\\3.5.33\\x86_64-pc-windows-gnu\\bundle.lock"
[2021-08-02T16:43:51.793961900+08:00][INFO][24220 bootstrapper_main_run_loop][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(507)] moving "C:\\Users\\26735\\.tabnine\\0.0.1\\x86_64-pc-windows-gnu\\TabNine-deep-cloud.exe" to "C:\\Users\\26735\\.tabnine\\3.5.33\\x86_64-pc-windows-gnu\\TabNine-deep-cloud.exe"
[2021-08-02T16:43:51.804453200+08:00][INFO][24220 bootstrapper_main_run_loop][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(507)] moving "C:\\Users\\26735\\.tabnine\\0.0.1\\x86_64-pc-windows-gnu\\TabNine-deep-local.exe" to "C:\\Users\\26735\\.tabnine\\3.5.33\\x86_64-pc-windows-gnu\\TabNine-deep-local.exe"
[2021-08-02T16:43:51.812086600+08:00][INFO][24220 bootstrapper_main_run_loop][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(507)] moving "C:\\Users\\26735\\.tabnine\\0.0.1\\x86_64-pc-windows-gnu\\TabNine.exe" to "C:\\Users\\26735\\.tabnine\\3.5.33\\x86_64-pc-windows-gnu\\TabNine.exe"
[2021-08-02T16:43:51.935821200+08:00][INFO][24220 bootstrapper_main_run_loop][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(507)] moving "C:\\Users\\26735\\.tabnine\\0.0.1\\x86_64-pc-windows-gnu\\WD-TabNine.exe" to "C:\\Users\\26735\\.tabnine\\3.5.33\\x86_64-pc-windows-gnu\\WD-TabNine.exe"
[2021-08-02T16:43:51.943546+08:00][INFO][24220 bootstrapper_main_run_loop][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(309)] finished extracting
[2021-08-02T16:43:51.946230500+08:00][WARN][24220 bootstrapper_main_run_loop][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(625)] relinquishing lock at path "C:\\Users\\26735\\.tabnine\\0.0.1\\x86_64-pc-windows-gnu\\bundle.lock"
[2021-08-02T16:43:51.983718800+08:00][WARN][24220 bootstrapper_main_run_loop][bootstrapper::bootstrapper_lib][Some("tabnine/bootstrapper/src/bootstrapper_lib.rs")][Some(314)] restart required
[2021-08-02T16:43:51.984834800+08:00][INFO][24220 bootstrapper_main_run_loop][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(222)] got kill message from bootstrapper. Bye Bye 👋🏻...

**Comment by maya-tabnine** [comment]
Thank you!
Can you also share your config file?
It is located at C:\Users\{UserName}\AppData\Roaming\TabNine\tabnine_config.json

**Comment by zz541843** [comment]
> Thank you!
> Can you also share your config file?
> It is located at C:\Users{UserName}\AppData\Roaming\TabNine\tabnine_config.json

{
  "version": "3.5.31",
  "hide_promotional_message": false,
  "beta_enabled": "No",
  "ignore_all_lsp": false,
  "creation_time": "2021-07-29T01:25:00.261251600Z",
  "guuid": "07af48da5bafd25eacce191cdff0be246d7ea5b2ed53d0cf1b0ac6e225848854",
  "semantic_status": {},
  "enable_telemetry": true,
  "user_understands_that_enabling_tabnine_cloud_sends_code_to_tabnine_servers": true,
  "hosted_deep_completions_enabled": "Disabled",
  "tabnine_cloud_host": null,
  "tabnine_cloud_certificate_domain": null,
  "tabnine_cloud_port": null,
  "cloud_whitelist": [],
  "num_of_suggestions": 5,
  "multiline_suggestions": null,
  "omit_prefix_suggestions": null,
  "api_key": null,
  "api_base_url": null,
  "binary_update_interval_seconds": null,
  "local_enabled": "Unset",
  "disable_local_when_using_battery": false,
  "hide_deep_information_message": false,
  "enable_power_saving_mode": false,
  "rate_limit_interval_seconds": null,
  "rate_limit_amount": null,
  "generation": 43,
  "local_model_size": null,
  "model_hash_override": null,
  "local_indexing": null,
  "heartbeat_interval_seconds": null,
  "last_service_level": null,
  "override_beams": null,
  "override_context_length": null,
  "onboarding": null,
  "has_git_repos": null,
  "gusr": {
    "dgc": "07af48da5bafd25eacce191cdff0be246d7ea5b2ed53d0cf1b0ac6e225848854"
  },
  "exclude_file_masks": null
}

**Comment by dimacodota** [comment]
@zz541843 can you please remove the `C:\Users\<user>\.tabnine\` folder and restart your IDE?
this should download the latest Tabnine binaries.
Thanks

**Comment by zz541843** [comment]
> @zz541843 can you please remove the `C:\Users\<user>\.tabnine\` folder and restart your IDE?
> this should download the latest Tabnine binaries.
> Thanks

Your words remind me that I searched all the files containing "tabnine" on this computer and deleted them. I downloaded tabnine again through the IDE and opened the C: \ users \ < user > \. Tabnine \ directory. There was only one 0kb file in it. After waiting for a long time, I finally had files such as tabnine.exe. I think the failure before was due to network problems
